diff --git a/src/FreeImage.rc b/src/FreeImage.rc
index 443858f..bc92a8d 100644
--- a/src/FreeImage.rc
+++ b/src/FreeImage.rc
@@ -6,8 +6,8 @@
 //
 
 VS_VERSION_INFO VERSIONINFO
- FILEVERSION 3,19,0,0
- PRODUCTVERSION 3,19,0,0
+ FILEVERSION 3,19,0,2
+ PRODUCTVERSION 3,19,0,2
  FILEFLAGSMASK 0x3fL
 #ifdef _DEBUG
  FILEFLAGS 0x1L
@@ -25,14 +25,14 @@ BEGIN
             VALUE "Comments", "FreeImage is an Open Source library project for developers who would like to support popular graphics image formats like PNG, BMP, JPEG, TIFF and others as needed by today's multimedia applications.\0"
             VALUE "CompanyName", "FreeImage\0"
             VALUE "FileDescription", "FreeImage library\0"
-            VALUE "FileVersion", "3, 19, 0, 0\0"
+            VALUE "FileVersion", "3, 19, 0, 2\0"
             VALUE "InternalName", "FreeImage\0"
             VALUE "LegalCopyright", "Copyright © 2003-2020 by FreeImage\0"
             VALUE "LegalTrademarks", "See http://freeimage.sourceforge.net\0"
             VALUE "OriginalFilename", "FreeImage.dll\0"
             VALUE "PrivateBuild", "\0"
             VALUE "ProductName", "FreeImage\0"
-            VALUE "ProductVersion", "3, 19, 0, 0\0"
+            VALUE "ProductVersion", "3, 19, 0, 2\0"
             VALUE "SpecialBuild", "\0"
         END
     END
